composer/loggers/object_store_logger.py

```diff
@@ -402,23 +415,32 @@ def post_close(self):
         if self._tempdir is not None:
             self._tempdir.cleanup()
 
-        # Reset all variables
-        self._tempdir = None
-        self._finished = None
-        self._workers.clear()
-        self._enqueued_objects.clear()
-        self._logged_objects.clear()
-        assert self._file_upload_queue.empty()
-
         # Empty the completed queue
+        # This cleanup will not be done by the enqueue_thread anymore, as that thread has been shut down
         while True:
             try:
-                self._completed_queue.get_nowait()
+                object_name = self._completed_queue.get_nowait()
             except queue.Empty:
                 break
+            self._enqueued_objects.remove(object_name)
             self._completed_queue.task_done()
 
+        if len(self._enqueued_objects) > 0 or len(self._logged_objects) > 0:
+            # Warn on all objects that have not been uploaded
+            object_names = list(self._enqueued_objects)
+            object_names.extend(self._logged_objects.keys())
+            warnings.warn(
+                RuntimeWarning('The following objects may not have been uploaded, likely due to a worker crash: ' +
+                               ', '.join(self._enqueued_objects)))
+
+        # Reset all variables
```

Comment:
Why is this necessary?

Reply:
This is needed if you are creating a new trainer with the same callback instance.